### PR TITLE
Fix Weights Tying Problem in TT-XLA Torch Infra

### PR DESCRIPTION
**Pull request by mmilosevicTT**
### Ticket
NA

### Problem description
Currently many training tests are failing on frontend issues related to missing gradients. After investigation it was noticed that `transformers` library uses feature of tied weights. This feature is well supported for CPU evaluation, but when it comes to TT, since we are using torch_xla, it is faulty. The problem is that torch_xla unties weights on model movement to device.

### What's changed
This PR introduces following changes:
- Call to `tie_weights` after moving to device for both CPU and TT. This is to ensure both runs properly use these weights.
- Update of training tests that were failing because of the problem with tied weights (`ttir.gather` was out of date, it was failing with the same problem as other tests).

### Checklist
- [x] Torch test infra fix

### CODE REVIEW

**Comment by AleksKnezevic** [comment]
@mmilosevicTT do you know why this happens? Are we sure it's in torch_xla vs open_xla vs our pjrt runtime?

File: tests/infra/testers/single_chip/model/torch_model_tester.py [hunk 1]
```diff
@@ -92,6 +93,38 @@ def __init__(
     def _configure_model(self) -> None:
         self._device_runner.set_training_mode(self._run_mode == RunMode.TRAINING)
         super()._configure_model()
+        
+        # Untie weights to avoid issues where CPU has less parameters than TT in memory.
+        self._model = self._untie_weights(self._model)
+    
+    def _untie_weights(self, model: torch.nn.Module) -> torch.nn.Module:
+        """
+        Detects all tied parameters in a HuggingFace model and unties them by
```
[the remark this answers is not in the document]
Comment (mmilosevicTT) [reply tests/infra/testers/single_chip/model/torch_model_tester.py]: Yes, they are different for tied version as well. The difference is that in tied version, we sum them up and put them in the embedding weights, while in untied version each is stored with its separate tensor.

So untied version has more memory, but more granularity in gradients. The problem comes when you try updating weights, which we do not do in testing infra.

**Review by sgligorijevicTT — APPROVED** [review]
